### PR TITLE
Add github ci config

### PR DESCRIPTION
**Pull request by manoranjith**
Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

### CODE REVIEW

**Comment by matthiasgeihs** [comment]
This looks fine, but the workflow is not triggered in this repository.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      run: |
+        make linkcheck
+        make html
+      #
+      # uses: maraskar/sphinx-action@master
+      # with:
+      #   # build-command: make linkcheck
+      #   pre-build-command: make linkcheck
+      #   build-command: make html
+      #   docs-folder: "./"
```
Comment (matthiasgeihs) [inline .github/workflows/build.yml:36]: This should be removed?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo bash setup.sh --quiet-mode
+
+    - name: add-git-credentials
+      run: |
+        git config --global user.name "circle-ci"
+        git config --global user.email "circle-ci@nomail.com"
```
Comment (matthiasgeihs) [inline .github/workflows/build.yml:24]: Do you need this? For what?

Comment (manoranjith) [reply .github/workflows/build.yml]: Nope, this has been removed in the current iteration. Instead the options in the corresponding github action is used.

File: .github/workflows/build.yml (hunk 1 above)
Comment (manoranjith) [reply .github/workflows/build.yml]: Yes, has been updated in the current iteration.

**Comment by manoranjith** [comment]
I was initially working on this from a fork on my repository. Hence, closing this before it is merged, in favor of #75.

**Pull request closed without merging**